### ISSUE TITLE
FOUC on initial load

### ISSUE THREAD

**Issue by emadabdulrahim** [modulz/stitches#169]
For a dark themed site, I have a minimal setup for a theme config and adding `background` to `body`

### Problem
I see flash of unstyled content on body, starting white and then dark.

**Deployed example**
https://stiches.vercel.app/

**Repo** 
https://github.com/emadabdulrahim/stitches-example

Should I be doing things differently?

**Comment by hadihallak** [comment]
Hey @emadabdulrahim 👋 
Thanks for submitting this issue. I would love to have a look but unfortunately, the repo link you provided seems to be broken.

but yeah I think this is a known issue that we also have on our official stitches website so we're definitely aiming to fix it

**Comment by emadabdulrahim** [comment]
@hadihallak The repo link should work now.

Thanks for taking a look

**Comment by peduarte** [comment]
I dont think this is specific to Stitches. It happens on SSR websites with dark mode, where dark mode is not the default theme.

Some more information:
- https://brianlovin.com/overthought/adding-dark-mode-with-next-js
- https://blog.maximeheckel.com/posts/switching-off-the-lights-part-2-fixing-dark-mode-flashing-on-servered-rendered-website

I need to fix this on here too https://github.com/modulz/stitches-site/issues/17 😄

**Comment by peduarte** [comment]
I'll close this issue, and you can follow https://github.com/modulz/stitches-site/issues/17 instead for the fix

**Comment by peduarte** [comment]
So it seems that this may be a legit bug. We need to validate it. 

The theory right now is that the theme CSS is not injected at the right time. 

More info in the issues on the stitches-site repo

**Comment by peduarte** [comment]
@biowaffeln maybe you can provide more information

**Comment by biowaffeln** [comment]
Hey, sorry for the late response!

I did some investigations of the bug a while ago, and the problem seems to be the following:

1. using the `noflash.js` script, a class name is applied to the body of the document before any Next.js code is executed. This seems to be working fine, i.e. the body has the dark theme class name applied to it.
2. Then, the css variables are injected as part of the `toString` method on the theme object, [see here](https://github.com/modulz/stitches/blob/a19bd6efde64bc1d6ec808e9d9276bbc3a338fe0/packages/core/src/index.ts#L296).
3. In a production build, however, the code for inserting the variables is only executed after the inital render, thus causing the flash of light mode. Here is a [link to a little demo](https://mega.nz/file/ZlQF3YCa#ks9pvAb_X9yt97O6Efu56NMtvqFPDFZDPc9iW1n3anM) demonstrating the effect.
4. I think that one solution could be to generate the theme styles as part of the styles returned from the `getStyles` method [here](https://github.com/modulz/stitches/blob/a19bd6efde64bc1d6ec808e9d9276bbc3a338fe0/packages/core/src/index.ts#L6802), so that it would get injected during the setup of `_document.tsx`. I tried debugging it, but unfortunately not very familiar with that part of the codebase. Perhaps @hadihallak @christianalfoni you guys want to take a look at it?
